unison-manual: Clean up cruft

gdt opened the pull request:
Drop mention of Windows 95.
Drop judgemental terms about TUI vs GUI.
Prune web links that are 404.
Don't use "32-bit" and "64-bit" as if they specify a CPU architecture. Drop content in favor of pointers to BUILDING.md and CONTRIBUTING.md.

Fixes #717

tleedjarv reviewed (commented):
I think the removals are good and in the right direction (but can remove even more) but I'm hesitant about some of the additions. In my view, with the current manual we don't need to focus on building from source or even on download instructions. It was a good idea when the HTML manual was available at the homepage. When that is no longer the case then the manual does not have to include anything about downloading and building from source.

gdt commented on doc/unison-manual.tex:
```diff
@@ -210,36 +193,23 @@
-If you don't see a pre-built executable for your architecture, you'll
-need to build it yourself.  See \sectionref{building}{Building Unison from Scratch}.
+See {\tt BUILDING.md} in the sources for instructions for building
+Unison for source.
```
heavily pruned

gdt commented on doc/unison-manual.tex:
```diff
@@ -185,19 +168,19 @@
-\SHOWURL{http://www.cis.upenn.edu/\home{bcpierce}/unison/faq.html}{Frequently Asked
-Questions page} helpful.  Pre-built binaries are available for a
-variety of platforms.
+Unison is designed to be easy to build from source.
```
pruned

gdt commented on doc/unison-manual.tex:
```diff
@@ -138,25 +138,8 @@
-tarball from the download page.  For some details on how the code is
-organized, etc., see the file {\tt CONTRIB}.
+See {\tt CONTRIBUTING.md} in the sources for information about
+contributing to Unison.
```
redone

gdt commented on doc/unison-manual.tex:
```diff
@@ -185,19 +168,19 @@
-\item a more sophisticated graphical interface, based on Gtk2 (on
-       Linux/Windows) or the native UI framework (on OSX).
+\item a graphical interface, based on GTK3 (on POSIX systems and Windows)
+       and alternatively based on macOS's UI framework.
```
pruned

gdt commented:
I just force-pushed a major rototill addressing your comments, which was mostly removing a lot of content that duplicates what is in sources/wiki, replaced by very very brief pointers.   I am not claiming I fixed everything you objected to, but if not please object again because I did not mean to non-respond to any comments.

tleedjarv commented on doc/unison-manual.tex:
```diff
@@ -132,31 +132,37 @@
 \URL{http://research.microsoft.com/\home{fournet}/}{Cedric Fournet}
 contributed to an even earlier prototype.
 
-\TOPSUBSECTION{Mailing Lists and Bug Reporting}{lists}
```
Removing this sections breaks a cross-ref in section "Going Further".

gdt replied:
fixed I think

tleedjarv commented on doc/unison-manual.tex:
```diff
@@ -132,31 +132,37 @@
+\ONEURL{https://github.com/bcpierce00/unison/wiki}
+\end{quote}
+
+\TOPSUBSECTION{Community, Maintenance, and Development}{development}
```
See how this section looks when rendered. Using `\paragraph`, like in the old contactsbody.tex, looks slightly better in my opinion.

gdt replied:
adjusted

tleedjarv commented on doc/unison-manual.tex:
```diff
@@ -183,69 +189,16 @@
   Synchronization}, and by University of Pennsylvania's Institute for
 Research in Cognitive Science (IRCS).
 
-\SECTION{Installation}{install}{install}
```
Removing this sections breaks a cross-ref in section "Local Usage".

gdt replied:
fixed I think

tleedjarv commented on doc/unison-manual.tex:
```diff
@@ -183,69 +189,16 @@
 
-Upgrading to a new version of Unison is as simple as throwing away the old
-binary and installing the new one.
+(This section is perhaps misplaced, but is early because it is far
+better to have at least skimmed it than to not know it exists.)
```
Agree, it feels misplaced. Perhaps make it a top-level section?

At some point (but not yet) we can remove all the text about compatibility and interoperability. Then the entire section becomes very short, or can be removed entirely.

gdt replied:
top it is

tleedjarv commented on doc/unison-manual.tex:
```diff
@@ -370,10 +305,10 @@
-this may require a little set-up (see \sectionref{click}{Click-starting
-  Unison}).  For this tutorial, we assume that you're starting it from the
-command line.
+The graphical version can also be run directly by clicking on its
+icon, but this may require a little set-up.
```
```suggestion
The graphical version can also be run directly by clicking on its icon.
```

gdt replied:
adopted

tleedjarv reviewed (commented):
I did a fresh review (I didn't go back to my old comments but I think you've covered everything).

There are some reference breaks, but otherwise it's good.

gdt commented:
This might actually be ready now.

tleedjarv commented:
Yes, better merge it. I feel like commit message is no longer accurate but I'll leave it to your judgement.

tleedjarv commented on doc/unison-manual.tex:
```diff
@@ -183,69 +197,11 @@
-\SUBSECTION{Downloading Unison}{download}
 
-See \ONEURL{https://github.com/bcpierce00/unison/wiki/Downloading-Unison}.
+\SECTION{Upgrading}{upgrading}
```
One bug still slipped through. `\SECTION` requires three arguments. The third argument may be left empty. A non-empty third argument seems to make this section a "topic" in the UI (seen by `unison -doc topic` and similarly in the GUI).
